From c02953ee70324e7a86bace49f5e117868d61b684 Mon Sep 17 00:00:00 2001
From: Leon Anavi <leon.anavi@konsulko.com>
Date: Wed, 9 Jul 2025 07:59:47 +0000
Subject: [PATCH] rpi: config: update default configuration for rpi5

Note BOOTCOMMAND is disabled in configuration since include/configs/rpi.h
defines it as part in commit "simplify booting and add dual rootfs".

USB is not started during preboot, since booting from USB is not supported
by default.

Upstream-Status: Inappropriate [Venus specific]

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>
---
 configs/rpi_arm64_defconfig | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/configs/rpi_arm64_defconfig b/configs/rpi_arm64_defconfig
index 9fe5d177943..abdabdb7891 100644
--- a/configs/rpi_arm64_defconfig
+++ b/configs/rpi_arm64_defconfig
@@ -3,6 +3,8 @@ CONFIG_POSITION_INDEPENDENT=y
 CONFIG_ARCH_BCM283X=y
 CONFIG_TARGET_RPI_ARM64=y
 CONFIG_ENV_SIZE=0x4000
+CONFIG_ENV_OFFSET=0x20000
+CONFIG_ENV_OFFSET_REDUND=0x24000
 CONFIG_DEFAULT_DEVICE_TREE="bcm2711-rpi-4-b"
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_DM_RESET=y
@@ -12,9 +14,10 @@ CONFIG_EFI_RUNTIME_UPDATE_CAPSULE=y
 CONFIG_EFI_CAPSULE_FIRMWARE_RAW=y
 CONFIG_BOOTSTD_DEFAULTS=y
 CONFIG_OF_BOARD_SETUP=y
+# CONFIG_USE_BOOTCOMMAND is not set
 CONFIG_FDT_SIMPLEFB=y
 CONFIG_USE_PREBOOT=y
-CONFIG_PREBOOT="pci enum; usb start;"
+CONFIG_PREBOOT="pci enum;"
 CONFIG_SYS_PBSIZE=1049
 # CONFIG_DISPLAY_CPUINFO is not set
 # CONFIG_DISPLAY_BOARDINFO is not set
@@ -27,7 +30,9 @@ CONFIG_CMD_PCI=y
 CONFIG_CMD_USB=y
 CONFIG_CMD_EFIDEBUG=y
 CONFIG_CMD_FS_UUID=y
-CONFIG_ENV_FAT_DEVICE_AND_PART="0:1"
+# CONFIG_ENV_IS_IN_FAT is not set
+CONFIG_ENV_IS_IN_MMC=y
+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
 CONFIG_TFTP_TSIZE=y
 CONFIG_DM_DMA=y
-- 
2.44.3

